awx_collection/test/awx/test_module_utils.py

```diff
@@ -36,41 +47,70 @@ def test_version_warning(collection_import, silence_warning):
     cli_data = {'ANSIBLE_MODULE_ARGS': {}}
     testargs = ['module_file2.py', json.dumps(cli_data)]
     with mock.patch.object(sys, 'argv', testargs):
-        with mock.patch('ansible.module_utils.urls.Request.open', new=mock_ping_response):
+        with mock.patch('ansible.module_utils.urls.Request.open', new=mock_awx_ping_response):
             my_module = TowerAPIModule(argument_spec=dict())
             my_module._COLLECTION_VERSION = "2.0.0"
-            my_module._COLLECTION_TYPE = "not-junk"
-            my_module.collection_to_version['not-junk'] = 'not-junk'
+            my_module._COLLECTION_TYPE = "awx"
             my_module.get_endpoint('ping')
     silence_warning.assert_called_once_with('You are running collection version 2.0.0 but connecting to tower version 1.2.3')
 
 
-def test_version_warning_strictness(collection_import, silence_warning):
+def test_version_warning_strictness_awx(collection_import, silence_warning):
     TowerAPIModule = collection_import('plugins.module_utils.tower_api').TowerAPIModule
     cli_data = {'ANSIBLE_MODULE_ARGS': {}}
     testargs = ['module_file2.py', json.dumps(cli_data)]
+    # Compare 1.0.0 to 1.2.3 (major matches)
     with mock.patch.object(sys, 'argv', testargs):
-        with mock.patch('ansible.module_utils.urls.Request.open', new=mock_ping_response):
+        with mock.patch('ansible.module_utils.urls.Request.open', new=mock_awx_ping_response):
             my_module = TowerAPIModule(argument_spec=dict())
             my_module._COLLECTION_VERSION = "1.0.0"
-            my_module._COLLECTION_TYPE = "not-junk"
-            my_module.collection_to_version['not-junk'] = 'not-junk'
+            my_module._COLLECTION_TYPE = "awx"
+            my_module.get_endpoint('ping')
+    silence_warning.assert_not_called()
+
+    # Compare 1.2.0 to 1.2.3 (major matches minor does not count)
+    with mock.patch.object(sys, 'argv', testargs):
+        with mock.patch('ansible.module_utils.urls.Request.open', new=mock_awx_ping_response):
+            my_module = TowerAPIModule(argument_spec=dict())
+            my_module._COLLECTION_VERSION = "1.2.0"
+            my_module._COLLECTION_TYPE = "awx"
+            my_module.get_endpoint('ping')
+    silence_warning.assert_not_called()
+
+def test_version_warning_strictness_tower(collection_import, silence_warning):
+    TowerAPIModule = collection_import('plugins.module_utils.tower_api').TowerAPIModule
+    cli_data = {'ANSIBLE_MODULE_ARGS': {}}
+    testargs = ['module_file2.py', json.dumps(cli_data)]
+    # Compare 1.2.0 to 1.2.3 (major/minor matches)
+    with mock.patch.object(sys, 'argv', testargs):
+        with mock.patch('ansible.module_utils.urls.Request.open', new=mock_tower_ping_response):
+            my_module = TowerAPIModule(argument_spec=dict())
+            my_module._COLLECTION_VERSION = "1.2.0"
```

Comment:
The line `my_module._COLLECTION_TYPE = "tower"` seems to be necessary here in order to actually silence the warning.